File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        s = set()
        l = 0
        n = len(nums)
        for r in range(n):
            if (r - l) > k:
                s.remove(nums[l])
                l += 1
            if nums[r] in s:
                return True
            s.add(nums[r])
        return False

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet;
        for(int i : nums)
            numSet.insert(i);
        int sequence = 0;
        for(int i : numSet)
        {
            int count = 1;
            if(!numSet.count(i-1))
            {
                while(numSet.count(i+count))
                    count++;
            }
            if(count>sequence) 
                sequence = count;
        }
        return sequence;
    }
};
